massa-protocol-worker/src/node_info.rs

```diff
@@ -53,10 +53,12 @@ impl NodeInfo {
             asked_blocks: Default::default(),
             connection_instant: Instant::now(),
             known_operations: Set::<OperationId>::with_capacity_and_hasher(
-                pool_settings.max_known_ops_size,
+                pool_settings.max_known_ops_size.saturating_add(1),
```

Comment:
I find this surprising, it would be better to keep the setting value as it is, and change the algorithm using the value. 

For example, instead of `while self.known_operations_queue.len() > max_ops_nb`, simply do `if self.known_operations_queue.len() == max_ops_nb`

Reply:
true, but changing to `if self.known_operations_queue.len() == max_ops_nb` would actually not solve the problem (the behavior is exactly the same) and would also require adding 1 to the capacity to ensure a no-realloc policy.

The best course in my opinion would be to add a comment to explain. Might do this in a followup

Reply:
> and would also require adding 1 to the capacity to ensure a no-realloc policy

How is that? The re-allocation happens not when reaching the capacity, but when exceeding it. So if after each insert you check that capacity has been reached, and if so, remove one, the set will not re-allocate(without needing to add 1).